packages/components/src/draggable/types.ts

```diff
@@ -45,6 +45,12 @@ export type DraggableProps = {
 	 * Arbitrary data object attached to the drag and drop event.
 	 */
 	transferData: unknown;
+	/**
+	 * Whether to stringify the transfer data type.
+	 *
+	 * @default true
+	 */
+	stringifyTransferData?: boolean;
```

Comment:
The result of `JSON.stringify` escapes double quotes which makes results with 'invalid' html, when we parse them after being dropped. This is mostly for back compat, in cases where someone would pass `string` data, to preserve the previous behavior.

Reply:
At first I had only a [check for a string](https://github.com/WordPress/gutenberg/pull/49297/commits/ba61ef9394276f09d07a09fff7cb46c754c646fa#diff-aedd48894c13d30fe5b3fe5e7501037e20c09b189d3ac2c537216adbce3b0af5R104). But if for some reason some consumers relied on that escaped quotes in their logic, it would break it. By having the prop we have the back compat.

Reply:
I'm not sure I see why would anyone rely on Draggable component escaping strings before they get into Droppable.

Reply:
TBH I don't either, but 🤷... I can revert to a simple check, what do you think?

Reply:
I'd prefer if we revert personally. 👍

Reply:
Could we instead just parse the JSON string in `onHTMLDrop` to prevent breaking changes?

Reply:
I think it could be 3. "both". That's the approach I took in the alternate PR https://github.com/WordPress/gutenberg/pull/49673. It's also the recommended way to set multiple fallbacks according to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API/Recommended_drag_types#dragging_images).

> text/HTML is a format that the browser understands that corresponds to "unserialized" HTML. We can't just put anything there. It's a bigger breaking change IMO.

The thing is we're already `JSON.stringify` everything putting in the `dataTransfer` prop. We could argue that it's guarded behind the `__experimentalTransferDataType` though and we never officially support passing other data types.

Reply:
Yeah, I think multiple data types is a good thing too but with our current code base (checking html first then, files then everything else), it has a bigger potential of disruption but I'm open to it. That said, I don't think it should be the block's responsibility to handle the dropping. `MediaPlaceholder` seems to be the right place. In fact I think we should deprecated `onHTMLDrop` prop of the media placeholder (we don't use it and it doesn't make sense for such component), instead we should handle it internally like we do for file dropping.